src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -74,6 +82,7 @@ public ElasticSearchService(
       DatasetServiceDAO datasetServiceDAO,
       StudyDAO studyDAO,
       LibraryCardDAO libraryCardDAO) {
+    this.executorService = Executors.newCachedThreadPool();
```

Comment:
CachedThreadPool and system performance can suffer from resource exhaustion if too many threads are created.

```suggestion
    this.executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
```

Reply:
This code should also probably hook JVM Shutdown and decide what to do with threads that are running (if you don't take the suggestion above) or are running/queued if you do.  :-)

Reply:
I lifted this from `SamDAO` so will make the same change there.